File: bench/string/1.go
package main

import (
	"fmt"
	"time"
	"io/ioutil"
	"strings"
)

func main() {
	var dat, err = ioutil.ReadFile("asset/lorem100mb")
	if err != nil {
		panic(err)
	}

	var start = time.Now()

	var result = strings.Split(string(dat), " ")
	fmt.Println(len(result))

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}

File: bench/string/1.rs
use std::fs::File;
use std::io::prelude::*;
use std::time::Instant;

fn main() -> std::io::Result<()> {
    let mut file = File::open("asset/lorem100mb")?;
    let mut contents = String::new();
    file.read_to_string(&mut contents)?;

    let start = Instant::now();

    let split = contents.split(" ");

    println!("{}", start.elapsed().as_nanos());

    let _vec: Vec<&str> = split.collect();
    Ok(())
}
